plugins/node/opentelemetry-instrumentation-aws-sdk/src/aws-sdk.ts

```diff
@@ -113,18 +114,21 @@ export class AwsInstrumentation extends InstrumentationBase<any> {
     ]);
 
     // patch for @smithy/middleware-stack for aws-sdk packages v3.363.0+
-    const v3SmithyMiddlewareStackFile = new InstrumentationNodeModuleFile(
-      '@smithy/middleware-stack/dist-cjs/MiddlewareStack.js',
-      ['>=1.0.1'],
-      this.patchV3ConstructStack.bind(this),
-      this.unpatchV3ConstructStack.bind(this)
-    );
+    const self = this;
     const v3SmithyMiddlewareStack = new InstrumentationNodeModuleDefinition(
       '@smithy/middleware-stack',
       ['>=2.0.0'],
-      undefined,
-      undefined,
-      [v3SmithyMiddlewareStackFile]
+      (moduleExports, moduleVersion) => {
+        const newExports = propwrap(
+          moduleExports,
+          'constructStack',
+          (orig: any) => {
+            self._diag.debug('propwrapping aws-sdk v3 constructStack');
+            return self._getV3ConstructStackPatch(moduleVersion, orig);
+          }
+        );
+        return newExports;
+      }
```

Comment:
Note for reviewers: There is now no *un*wrapping. I'm not sure if that has possible adverse effects.

Reply:
I have to admit i'm not as knowledgeable on this as I'd like. IIUC we use the unwrapping mostly for testing and ensuring we don't double wrap a library (using auto-instrumentations and aws-sdk instrumentation directly and registering the same one twice, or registering it twice, but with different instrumentation versions). It could also be that users want to disable the instrumentation on runtime for some reason (though, I've never seen such a thing done IRL). 

Hoping that maybe @blumamir or @dyladan have more insights.

Reply:
This was discussed in the OTel JS SIG today (https://docs.google.com/document/d/1tCyoQK49WVcE-x8oryZOTTToFm7sIeUhxFPm9g-qL1k/edit#heading=h.7y4kcwttgzlh). Dan suggested not having an unwrap would be fine. In his opinion (I hope I'm not misrepresenting), instrumentation unwrap functionality in general is really there for testing and probably isn't something to rely heavily on for production usage.